src/tests/system/tests/test_cache.py

```diff
@@ -146,70 +146,174 @@ def test_cache__writes_to_both_database_files_when_using_fully_qualified_names(
+):
+    """
+    :title: Check latest group changes are reflected when modified and deleted
+    :setup:
+        1. Create groups 'group1' and group2'
+        2. Start SSSD
+        3. Lookup groups
+    :steps:
+        1. Create and add 'user1' to 'group1' and delete 'group2' and clear cache
+        2. Lookup group 'group1'
+        3. Lookup group 'group2'
+    :expectedresults:
+        1. User 'user1' is created and added to group 'group1'  and group 'group2' is deleted
+        2. Group 'group1' is found and contains 'user1'
+        3. Group 'group2' is not found
+    :customerscenario: False
+    """
+    group1 = provider.group("group1").add()
+    group2 = provider.group("group2").add()
+    client.sssd.start()
+
+    assert client.tools.getent.group("group1"), "Group group1 not found!"
+    assert client.tools.getent.group("group2"), "Group group2 not found!"
+
+    group1.add_member(provider.user("user1").add())
+    group2.delete()
+
+    client.sssctl.cache_expire(everything=True)
+
+    assert client.tools.getent.group("group1"), "Group group1 not found!"
+    assert "user1" in str(client.tools.getent.group("group1")), "Group group1 is missing a user!"
+    assert not client.tools.getent.group("group2"), "Group group2 found!"
+
+
+@pytest.mark.integration
+@pytest.mark.importance("low")
+@pytest.mark.parametrize("obj", ["user", "group", "netgroup"])
+@pytest.mark.parametrize("dbs", ["cache", "timestamps"])
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_cache__invalidate_entries_in_domain_and_timestamps_caches(
+    client: Client,
+    provider: LDAP,
+    obj: str,
+    dbs: str,
+):
+    """
+    :title: Invalidates object entries in the domain and timestamps caches using dataExpireTimestamp attribute
+    :setup:
+        1. Create user, group and netgroup
+        2. Add user as a member to group and netgroup
+        3. Start SSSD and lookup user, group and netgroup
+    :steps:
+        1. Search for object attribute dataExpireTimestamp in domain and timestamps ldb caches
+        2. Clear the SSSD cache for the object
+        3. Search for the object attribute dataExpireTimestamp in domain and timestamps ldb caches
+    :expectedresults:
+        1. Attribute is found
+        2. Cache is cleared for the object
+        3. Attribute is found and the value is '[1]' meaning it is cleared
+    :customerscenario: False
+    """
+    # Unable to parametrize provider method when adding objects, so all object types are created
+    user = provider.user("user").add()
+    provider.group("group").add().add_member(user)
+    provider.netgroup("netgroup").add().add_member(user=user)
+    client.sssd.start()
+
+    assert client.tools.getent.passwd("user")
+    assert client.tools.getent.group("group")
+    assert client.tools.getent.netgroup("netgroup")
+
+    path = f"/var/lib/sss/db/{dbs}_{client.sssd.default_domain}.ldb"
+
+    # netgroups does not use the fully qualified name and does not use the timestamp cache
```

Comment:
From a quick glance, netgroups /should/ be using ts cache as well:
`sysdb_search_netgroup_by_name()` -> `sysdb_search_entry()` -> `sysdb_cache_search_entry()` + `sysdb_merge_msg_list_ts_attrs()`